controllers/extensions/addon_controller_test.go

```diff
@@ -630,6 +630,33 @@ var _ = Describe("Addon controller", func() {
 			})
 			addonStatusPhaseCheck(2, extensionsv1alpha1.AddonFailed, nil)
 		})
+
+		It("should have provider and version in spec for Addon with provider and version in label initially", func() {
+			By("By create an addon with auto-install and labels contains provider and version information")
+			createAddonSpecWithRequiredAttributes(func(newOjb *extensionsv1alpha1.Addon) {
+				newOjb.Labels = map[string]string{
+					"addon.kubeblocks.io/provider": "apecloud",
+					"addon.kubeblocks.io/version":  "0.9.0",
+				}
+				newOjb.Spec.Installable.AutoInstall = true
+			})
+
+			By("By addon autoInstall auto added")
+			enablingPhaseCheck(2)
+
+			By("By enabled addon with fake completed installation job status")
+			fakeInstallationCompletedJob(2)
+
+			By("By setting the provider and version in spec for Addon")
+			Eventually(func(g Gomega) {
+				_, err := doReconcile()
+				g.Expect(err).To(Not(HaveOccurred()))
+				addon = &extensionsv1alpha1.Addon{}
+				g.Expect(testCtx.Cli.Get(ctx, key, addon)).To(Not(HaveOccurred()))
+				g.Expect(addon.Spec.Provider).Should(Equal("apecloud"))
+				g.Expect(addon.Spec.Version).Should(Equal("0.9.0"))
+			}).Should(Succeed())
+		})
 	})
```

Comment:
we may add one more test case, where Addon is created, but not 'Enabled', and make sure provider and version info are set.